tools/eclipse_prj_helper/Makefile.template

```diff
@@ -0,0 +1,47 @@
+###########################################################################
+#
+#  Copyright (c) 2013-2017, ARM Limited, All Rights Reserved
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+###########################################################################
+
+###########################################################################
+#            DO NOT EDIT - This is an auto-generated file                 #
+###########################################################################
+# generated from:
+#   $template_file
+
+
+WORKSPACE_DIR := $workspace_dir
+UVISOR_ROOT_DIR := $uvisor_dir
+IMPORTER_DIR := $importer_dir
+
+PLATFORM := $target
+TOOLCHAIN := $toolchain
+
+
+
+.PHONY: all clean
+all:
+	@echo Build uVisor
+	@test -e $(UVISOR_ROOT_DIR) && $(MAKE) -C $(UVISOR_ROOT_DIR)
+	@echo Publish uVisor build artifacts
+	@test -e $(UVISOR_ROOT_DIR) && $(MAKE) -C $(IMPORTER_DIR) publish
+	@echo Build Workspace
+	@cd $(WORKSPACE_DIR) && mbed compile --profile debug -m $(PLATFORM) -t $(TOOLCHAIN)
```

Comment:
Aside: Calling make in make is generally a sign that something has gone horribly wrong with your makefile.